templates/tips_and_tricks.md

```diff
@@ -0,0 +1,97 @@
+It is currently focussed around Lean 3.
+
+### Commenting out large proofs using `sorry`
+
+Sometimes skipping subportions of a proof is helpful (e.g. to disable a portion which is slow while working on the rest). The straightforward way to do so is to comment out the lines which constitute the subportion, but if a subgoal has been surrounded by braces, you can also temporarily skip it by prepending `sorry;`
+```
+sorry; { rw blah,
+  ...
+  HUGE PROOF,
+  ... }
+```
+With this method only one line needs to be changed, syntax highlighting is usually preserved, and jumping to definition still works. It also has the advantage that it is extremely easy to remove all instances of `sorry;` from a file when finished using find and replace, and to quickly toggle this on and off to check things still work when desired.
+
+### Memory and deterministic timeout
+Lean has two configurable flags for configuring its resource usage, memory and execution time.
+While the defaults for these are usually fairly sensible, when writing proofs it can be helpful to increase these options, if deterministic timeouts or out of memory issues slow down development.
+This is especially helpful when using tactics like [`library_search`](/mathlib_docs/tactics.html#library_search), which may run for a long time but then produce a result that tells us a much faster tactic to use if they succeed.
+
+To change these settings we can add the flags `-M <some number of megabytes>` (default 4096) for the memory limit and `-T <number of allocations>` (default 100000).
+In VSCode this can be configured by opening the settings and searching for `Lean Memory Limit` and `Lean Time Limit`.
+Mathlib itself enforces a deterministic timeout of 100000, and often tries to have individual tactic calls use much less, so it is good practice not to rely on a large value for this setting.
+
+
+### Old mode
+If you change a file early on in the import hierarchy and then attempt to work on a file that depends on it much later, then Lean will try to check all files in between to ensure they didn't break.
+This behaviour can cause the Lean server to become unresponsive, or simply slow down development.
+You can call lean with the "--old" flag to help in these situations, though it does come with some caveats.
+In VSCode you can enable it by opening the VSCode settings and searching for "Lean: extra options" then add the flag `--old`.
+For Emacs the variable `lean-extra-arguments` can be modified to add this flag.
+This causes lean not to recompile oleans for unedited files and just assume everything worked, so in this situation it is helpful to avoid recompilation.
+However you do then run the risk that your change broke or changed the meaning of some file in-between. Thus it most applicable when you are adding a new lemma/definition early in the hierarchy that you are sure won't break later files, rather than when you are changing existing definitions in an incompatible way.
```

Comment:
```suggestion
However you do then run the risk that your change broke or changed the meaning of some file in-between. Thus this suggestion mostly applies when you are adding a new lemma/definition early in the hierarchy that you are sure won't break later files, rather than when you are changing existing definitions in an incompatible way.
```